src/test/java/com/exadel/movie_new/tests/MovieServiceTest.java

```diff
@@ -0,0 +1,158 @@
+package com.exadel.movie_new.tests;
+
+import com.exadel.movie_new.dao.MovieDaoImpl;
+import com.exadel.movie_new.exception.MovieAlreadyExists;
+import com.exadel.movie_new.exception.MovieNotFoundException;
+import com.exadel.movie_new.model.Movie;
+import com.exadel.movie_new.service.MovieServiceImpl;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+
+
+@SpringBootTest
+public class MovieServiceTest {
+
+    @InjectMocks
+    private MovieServiceImpl movieService;
+
+    @Mock
+    private MovieDaoImpl movieDaoImplMock;
+
+    private final Movie newMovie;
+    private final Movie newMovie2;
+
+    {
+        newMovie = new Movie();
+        newMovie.setId("100");
+        newMovie.setTitle("Kill Bill");
+        newMovie.setGenres(List.of("Comedy", "Romance"));
+
+        newMovie2 = new Movie();
+        newMovie2.setId("101");
+        newMovie2.setTitle("Anastazja");
+        newMovie2.setGenres(List.of("Comedy", "Romance"));
+    }
+
+    @Test
+    @DisplayName("Should save the movie to a file")
+    void shouldAddNewMovie() {

```

Comment:
This method looks like redundant because in **addMovie** method you return exactly the same movie as you set in method (without any manipulation with it). In case your movieDao.write() could return saved movie - then it could be ok, but for now you pass movie, call **write()** and return the same movie. No reason for such logic. I recommend you update **addMovie** (it could be void) And remove this test